pkg/cmd/util/cmd.go

```diff
@@ -80,31 +81,44 @@ func convertItemsForDisplay(objs []runtime.Object, preferredVersions ...unversio
 
 	for i := range objs {
 		obj := objs[i]
-		kind, _, err := kapi.Scheme.ObjectKind(obj)
-		if err != nil {
-			return nil, err
-		}
-		groupMeta, err := registered.Group(kind.Group)
+		kinds, _, err := kapi.Scheme.ObjectKinds(obj)
 		if err != nil {
 			return nil, err
 		}
 
-		requestedVersion := unversioned.GroupVersion{}
-		for _, preferredVersion := range preferredVersions {
-			if preferredVersion.Group == kind.Group {
-				requestedVersion = preferredVersion
-				break
+		// Gather all groups where the object kind is known.
+		groups := []*apimachinery.GroupMeta{}
+		for _, kind := range kinds {
+			groupMeta, err := registered.Group(kind.Group)
+			if err != nil {
+				return nil, err
 			}
+			groups = append(groups, groupMeta)
 		}
 
 		actualOutputVersion := unversioned.GroupVersion{}
-		for _, externalVersion := range groupMeta.GroupVersions {
-			if externalVersion == requestedVersion {
-				actualOutputVersion = externalVersion
-				break
+		// Find the first preferred version that contains the object kind group.
+		// If there are more groups for the given resource, prefer those that are first in the
+		// list of preferred versions.
+		for _, version := range preferredVersions {
```

Comment:
@smarterclayton this will fix the import compose case, so we don't have to change the test... also by accident this also fixes `--output-version` so you can pass multiple versions and the order will be honored... For example:

```console
# oc import docker-compose -f \
   test/testdata/app-scenarios/docker-compose/complex/docker-compose.yml -o yaml \
   --output-version=image.openshift.io/v1,v1 | grep apiVersion
apiVersion: v1
- apiVersion: image.openshift.io/v1
- apiVersion: v1
- apiVersion: image.openshift.io/v1
- apiVersion: v1
- apiVersion: image.openshift.io/v1
- apiVersion: v1
- apiVersion: v1
- apiVersion: image.openshift.io/v1
- apiVersion: v1
- apiVersion: v1
- apiVersion: v1
```

This way I can change the preferred version to be `image.openshift.io/v1` over the legacy.
If you don't specify the `--output-version`, we default to prefer the `/v1` (aka legacy) API, so nothing will break in term of backward compatibility....